src/utils.js

```diff
@@ -79,3 +80,65 @@ export const convertBufferToLines = (current, previous) => {
       index !== lastNewlineIndex ? buffer.slice(lastNewlineIndex) : null,
   };
 };
+
+export const getLinesLengthRanges = rawLog => {
+  const { length } = rawLog;
+  const linesRanges = [];
+  let lastNewlineIndex = 0;
+  let index = 0;
+
+  while (index < length) {
+    const current = rawLog[index];
+    const next = rawLog[index + 1];
+
+    if (isNewline(current, next)) {
+      lastNewlineIndex =
+        current === ENCODED_CARRIAGE_RETURN && next === ENCODED_NEWLINE
+          ? index + 2
+          : index + 1;
+
+      index = lastNewlineIndex;
+      linesRanges.push(index);
+    } else {
+      index += 1;
+    }
+  }
+
+  return linesRanges;
+};
+
+export const searchFormatPart = ({
+  searchKeywords,
+  nextFormatPart,
+  replaceJsx,
+}) => part => {
+  let formattedPart = part;
+
+  if (nextFormatPart) {
+    formattedPart = nextFormatPart(part);
+  }
+
+  if (part.indexOf(searchKeywords) > -1) {
+    return reactStringReplace(formattedPart, searchKeywords, replaceJsx);
+  }
+
+  return formattedPart;
+};
+
+export const concatenateUint8Arrays = (...arrays) => {
```

Comment:
I already have a buffer concat method in this file called `bufferConcat`, but it operates on only 2 buffers. Do you foresee where that is insufficient, or an instance where we operate on more than 2 buffers?